### PR TITLE
Update UIKit, switch to sass-embedded and make new-theme compatible with node 16, 18 and 19

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | We want to switch to sass-embedded to avoid some build problems on different environment and improve performances, at the same time we want to update some webpack dependencies to make it compatible and more up-to-date
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | new-theme should build properly and migrated pages should work as expected, please check for UIKit related things and styles
| Possible impacts? | new-theme build

### CODE REVIEW

**Review by matks — DISMISSED** [review]

File: admin-dev/themes/new-theme/.webpack/common.js [hunk 1]
```diff
@@ -363,17 +364,17 @@ module.exports = {
       // FILES
       {
         test: /.(jpg|png|woff2?|eot|otf|ttf|svg|gif)$/,
-        loader: 'file-loader',
-        options: {
-          name: '[hash].[ext]',
+        type: 'asset/resource',
+        generator: {
+          filename: '[hash].[ext]'
```
Comment (matthieu-rolland) [inline admin-dev/themes/new-theme/.webpack/common.js]:
```suggestion
          filename: '[hash].[ext]',
```

File: admin-dev/themes/new-theme/.webpack/common.js [hunk 2]
```diff
@@ -363,17 +364,17 @@ module.exports = {
...
         exclude: /MaterialIcons-Regular\.(woff2?|ttf)$/,
       },
       {
         test: /MaterialIcons-Regular\.(woff2?|ttf)$/,
-        loader: 'file-loader',
-        options: {
-          name: '[hash].preload.[ext]',
+        type: 'asset/resource',
+        generator: {
+          filename: '[hash].preload.[ext]'
```
Comment (matthieu-rolland) [inline admin-dev/themes/new-theme/.webpack/common.js]:
```suggestion
          filename: '[hash].preload.[ext]',
```

**Review by matthieu-rolland — COMMENTED** [review]
Let's make eslint happy, lgtm otherwise :+1:

**Review by zuk3975 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by paulnoelcholot** [comment]
Hello @NeOMakinG,

I have tested your PR on develop and I found bugs.
On right with the PR, on left without the PR:

1. Missing text : 
On Catalog > Stock
image

2. Unupgradable module
On Module>Module Manager> Alerts (or) Updates
image

3. Empty search after changing the language
https://user-images.githubusercontent.com/100699445/208868467-87a999fd-3b8b-45ff-a196-81951e35398f.webm

**Comment by Hlavtox** [comment]
@paulnoelcholot Are you sure you have ps_distributionapi installed and working? Not sure if a label could magically disappear.

Snímek obrazovky 2022-12-21 103316

Regarding the missing text, if you have checked develop, you would see that it's also missing.
Snímek obrazovky 2022-12-21 103507

**Comment by NeOMakinG** [comment]
@paulnoelcholot could you check on develop if you can reproduce these bugs? I think they are out of scope of this PR but related to the Vue 2 to Vue 3 upgrade, it would deserve a new issue and another PR to fix them

**Comment by paulnoelcholot** [comment]
@NeOMakinG, I tested on develop and the problem comes from develop and not from PR for bugs 1 and 2. Thanks.
On the other hand, problem 3 is indeed linked to PR

**Comment by NeOMakinG** [comment]
> @NeOMakinG, I tested on develop and the problem comes from develop and not from PR for bugs 1 and 2. Thanks. On the other hand, problem 3 is indeed linked to PR

@paulnoelcholot I will give it a look but I'm 99% sure it's linked to https://github.com/PrestaShop/PrestaShop/pull/28463 as this PR doesn't adds anything functional

**Comment by NeOMakinG** [comment]
@paulnoelcholot should be good but we have a problem on develop and other branches

If you remove the `package-lock.json` and `npm i && npm run build`, you'll have the same behavior your noticed on that PR, that's weird

This PR is updated and should pass now but we should investigate on that issue outside of this scope

**Comment by NeOMakinG** [comment]
@paulnoelcholot Finally I fixed it directly inside this PR, should be good now

**Review by matks — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by florine2623** [comment]
Automated tests ✅  : https://github.com/florine2623/testing_pr/actions/runs/3847056202

**Review by florine2623 — APPROVED** [review]
Hello @NeOMakinG ,

[Bug 3](https://github.com/PrestaShop/PrestaShop/pull/29842#issuecomment-1361055879) mentioned by @paulnoelcholot is fixed !

No other bug found.

It is QA ✅ 
Thanks!

**Pull request merged**